docs/datamodeling.rst

```diff
@@ -0,0 +1,330 @@
+            } catch(EdgeDBException err) { // deserialization error
+            
+            } catch(OperationNotSupportedException err) { // read/IO error
+
+            }
+        }
+    }
+
+This approach isn't viable for large data structure maps. Instead, it is useful 
+for other data type representations, like tuples:
+
+.. code-block:: java
+
+        @EdgeDBDeserializer
+        public SimpleTuple(ObjectEnumerator enumerator) 
+        throws EdgeDBException, OperationNotSupportedException {
+            elements = new ArrayList<>();
+
+            while(enumerator.hasRemaining()) {
+                var enumerationElement = enumerator.next();
+
+                assert enumerationElement != null;
+
+                elements.add(new Element(
+                    enumerationElement.getType(), 
+                    enumerationElement.getValue()
+                ));
+            }
+        }
+
+Value consumers
+^^^^^^^^^^^^^^^
+
+Value consumers take in the fields' values in the constructor, mapped by a 
+``@EdgeDBName`` annotation:
+
+.. tabs::
+
+    .. code-tab:: java
+        :caption: Java
+
+        @EdgeDBType
+        public class Person {
+            private final String name;
+            private final int age;
+
+            @EdgeDBDeserializer
+            public Person(
+                @EdgeDBName("name") String name,
+                @EdgeDBName("age") int age
+            ) {
+                this.name = name;
+                this.age = age;
+            }
+        }
+
+    .. code-tab:: sdl
+        :caption: Schema
+
+        module default {
+            type Person {
+                property name -> str;
+                property age -> int32;
+                multi link friends -> Person;
+            }
+        }
+
+
+Polymorphic types
+-----------------
+
+The binding supports polymorphic types, allowing you to reflect your abstract 
+schema types in code. For example:
+
+.. tabs::
+
+    .. code-tab:: java
+        :caption: Java
+
+        @EdgeDBType
+        public abstract class Media {
+            public String title;
+        }
+
+        @EdgeDBType
+        public class Show extends Media {
+            public Long seasons;
+        }
+
+        @EdgeDBType
+        public class Movie extends Media {
+            public Long release_year;
+        }
+
+    .. code-tab:: sdl
+        :caption: Schema
+
+        module default {
+            abstract type Media {
+                required property title -> str {
+                    constraint exclusive;
+                }
+            }
+        
+            type Movie extending Media {
+                required property release_year -> int64;
+            }
+        
+            type Show extending Media {
+                required property seasons -> int64;
+            }
+        }
+
+You can then specify ``Media`` as a result of a query; the binding will
+then discover any subclasses of ``Media`` and deserialize the subclasses as a 
+result.
```

Comment:
Call me out if I've changed the meaning here. Was trying to reduce density of "then."

```suggestion
With this schema, you can specify ``Media`` as a result of a query. The binding
will then discover any subclasses of ``Media`` and deserialize the subclasses
as a result.
```

Reply:
lgtm